### PR TITLE
MajorVersion and MinorVersion is coming from external input and it ma…

### PR DESCRIPTION
**Pull request by Zhiqiang520**
…ybe greater than ten.
fix: #699
Signed-off-by: Zhao, Zhiqiang <zhiqiang.zhao@intel.com>

### CODE REVIEW

**Comment by jyao1** [comment]
Would you please clarify the call stack for this error?

How the version is propagated from external to this libspdm_create_signing_context()?

I think checking version should be the first step for every command/response...

**Comment by Zhiqiang520** [comment]
The call stack for this error is as following:
```
    #4 0x56384c in libspdm_debug_assert /home/test/libspdm/os_stub/debuglib/debuglib.c:43:9
    #5 0x58fdad in libspdm_create_signing_context /home/test/libspdm/library/spdm_crypt_lib/libspdm_crypt_crypt.c:1620:5
    #6 0x592f8a in libspdm_req_asym_sign /home/test/libspdm/library/spdm_crypt_lib/libspdm_crypt_crypt.c:3048:9
    #7 0x59eaa0 in libspdm_requester_data_sign /home/test/libspdm/os_stub/spdm_device_secret_lib_sample/lib.c:739:18
    #8 0x561b74 in libspdm_test_responder_finish_case8 /home/test/libspdm/unit_test/fuzzing/test_responder/test_spdm_responder_finish_rsp/finish_rsp.c:387:5
    #9 0x56233c in libspdm_run_test_harness /home/test/libspdm/unit_test/fuzzing/test_responder/test_spdm_responder_finish_rsp/finish_rsp.c:433:5
```

**Comment by jyao1** [comment]
OK. The issue is at 
 #8 0x561b74 in libspdm_test_responder_finish_case8 /home/test/libspdm/unit_test/fuzzing/test_responder/test_spdm_responder_finish_rsp/finish_rsp.c:387:5

```
        m_spdm_finish_request.header.spdm_version << SPDM_VERSION_NUMBER_SHIFT_BIT, SPDM_FINISH,
```

You cannot input fuzzed version. It is illegal configuration.

File: library/spdm_crypt_lib/libspdm_crypt_crypt.c [hunk 1]
```diff
@@ -1617,8 +1617,10 @@ libspdm_create_signing_context (
     LIBSPDM_ASSERT((spdm_version >> SPDM_VERSION_NUMBER_SHIFT_BIT) > SPDM_MESSAGE_VERSION_11);
 
     /* So far, it only leaves 1 bytes for version*/
-    LIBSPDM_ASSERT((((spdm_version >> 12) & 0xF) < 10) &&
-                   (((spdm_version >> 8) & 0xF) < 10));
+    if ((((spdm_version >> 12) & 0xF) >= 10) ||
```
Comment (jyao1) [inline library/spdm_crypt_lib/libspdm_crypt_crypt.c:1620]: This is internal function.

The caller need guarantee the version is correct.

**Review by jyao1 — CHANGES_REQUESTED** [review]
Please make sure the caller is correct.

File: unit_test/fuzzing/test_responder/test_spdm_responder_finish_rsp/finish_rsp.c [hunk 2]
```diff
@@ -384,6 +384,11 @@ void libspdm_test_responder_finish_case8(void **State)
     /* session_transcript.message_k is 0*/
     libspdm_append_managed_buffer(&th_curr, req_cert_buffer_hash, hash_size);
     libspdm_append_managed_buffer(&th_curr, &m_spdm_finish_request, sizeof(spdm_finish_request_t));
+    /* The caller need guarantee the version is correct, both of MajorVersion and MinorVersion should be less than 10.*/
+    if (((m_spdm_finish_request.header.spdm_version & 0xF) >= 10) ||
+        (((m_spdm_finish_request.header.spdm_version >> 4) & 0xF) >= 10)) {
+        m_spdm_finish_request.header.spdm_version = SPDM_MESSAGE_VERSION_11;
```
Comment (jyao1) [inline unit_test/fuzzing/test_responder/test_spdm_responder_finish_rsp/finish_rsp.c:390]: I recommend to change to VERSION_12.

**Review by jyao1 — APPROVED** [review]

**Comment by jyao1** [comment]
Merge, since it is test code only.

**Pull request merged**